src/test/java/io/vertx/core/spi/executor/Utils.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.spi.executor;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.concurrent.ExecutorService;
+
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+import javax.tools.ToolProvider;
+
+import org.junit.rules.ExternalResource;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.test.core.LogContainsRule;
+
+public class Utils {
```

Comment:
ServiceHelperTest is using also Java compiler to do something similar, can you update the ServiceHelperTest to use this class instead?

Reply:
Yes OK will do, probably tomorrow now.

Reply:
Having looked at 82e0c4d more closely, does the above still stand or are you going to backport
what master does?

Reply:
I chose that it was part of the test to move SPI service classes on and off the classpath in the tests to try to make them more realistic - that is why I did not have them compiled and on the classpath all at once. However, its true that the ServiceLoader provider configuration files alone could be used and it is not our job to test Java ServiceLoading. I copied the style and code from one that Clement wrote.

@vietj What would you like me to do:

1. backport 82e0c4d and eliminate the compilation from the tests in my PR
2. as mentioned above, make 3.9 ServiceHelper and my PR tests use a common Utils method (maybe relocated out of executor/Utils to a more generic location)
3. nothing, as you are going to backport 82e0c4d at some point
?

Reply:
option 1. is my favorite, it's a relevant cleanup

Reply:
That's fine by me. I will learn more as it was doing mnv stuff I have not done that way before
and I agree it is more efficient not to compile 'in' the test.
I will push another commit on Monday.

Reply:
Done.